src/simplewallet/simplewallet.cpp

```diff
@@ -2504,16 +2504,25 @@ bool simple_wallet::transfer_main(int transfer_type, const std::vector<std::stri
   size_t fake_outs_count = 0;
   if(local_args.size() > 0) {
     size_t ring_size;
-    if(!epee::string_tools::get_xtype_from_string(ring_size, local_args[0]))
+    float ring_size_faulty;
```

Comment:
This is just testing for an amount first. What if the user puts the payment id first instead, or mispells the priority ? If you object to the original error messages, then I think it's better to have a standard error message which displays the full syntax. Simpler code, and it gets all cases of incorrectness.